contracts/external/gmxV2/GmxV2IsolationModeWrapperTraderV2.sol

```diff
@@ -0,0 +1,361 @@
+            );
+
+            bytes32 depositKey = exchangeRouter.createDeposit(depositParamsTest);
+            _setDepositInfo(depositKey, DepositInfo(tradeOriginatorForStackTooDeep, accountNumber));
+            emit DepositCreated(depositKey);
+        }
+
+        IGmxV2IsolationModeVaultFactory(address(VAULT_FACTORY())).setIsVaultFrozen(
+            tradeOriginatorForStackTooDeep,
+            true
+        );
+        IGmxV2IsolationModeVaultFactory(address(VAULT_FACTORY())).setShouldSkipTransfer(
+            tradeOriginatorForStackTooDeep,
+            true
+        );
+        return _minOutputAmount;
+    }
+
+    function _depositOtherTokenIntoDolomiteMarginFromTokenConverter(
+        address _token,
+        uint256 _amount,
+        DepositInfo memory _info,
+        IGmxV2IsolationModeVaultFactory factory
+    ) internal {
+        IERC20(_token).safeApprove(address(factory), _amount);
+        factory.depositOtherTokenIntoDolomiteMarginFromTokenConverter(
+            _info.vault,
+            _info.accountNumber,
+            DOLOMITE_MARGIN().getMarketIdByTokenAddress(_token),
+            _amount
+        );
+    }
+
+    function _setCallbackGasLimit(uint256 _callbackGasLimit) internal {
+        _setUint256(_CALLBACK_GAS_LIMIT_SLOT, _callbackGasLimit);
+    }
+
+    function _setSlippageMinimum(uint256 _slippageMinimum) internal {
+        Require.that(
+            _slippageMinimum < _SLIPPAGE_BASE && _slippageMinimum > 0,
+            _FILE,
+            "Invalid slippageMinimum"
+        );
+        _setUint256(_SLIPPAGE_MINIMUM_SLOT, _slippageMinimum);
+    }
+
+    function _setDepositInfo(bytes32 _key, DepositInfo memory _info) internal {
+        DepositInfo storage storageInfo = _getDepositSlot(_key);
+        storageInfo.vault = _info.vault;
+        storageInfo.accountNumber = _info.accountNumber;
+    }
+
+    function _approveIsolationModeTokenForTransfer(
+        address _vault,
+        address _receiver,
+        uint256 _amount
+    )
+    internal
+    override {
+        VAULT_FACTORY().enqueueTransferIntoDolomiteMargin(_vault, _amount);
+        IERC20(address(VAULT_FACTORY())).safeApprove(_receiver, _amount);
+    }
+
+    function _checkSlippage(address _inputToken, uint256 _inputAmount, uint256 _minOutputAmount) internal view {
+        IDolomiteMargin dolomiteMargin = DOLOMITE_MARGIN();
+
+        IDolomiteStructs.MonetaryPrice memory inputPrice = dolomiteMargin.getMarketPrice(
+            dolomiteMargin.getMarketIdByTokenAddress(_inputToken)
+        );
+        IDolomiteStructs.MonetaryPrice memory outputPrice = dolomiteMargin.getMarketPrice(
+            dolomiteMargin.getMarketIdByTokenAddress(address(VAULT_FACTORY()))
+        );
+        uint256 inputValue = _inputAmount * inputPrice.value;
+        uint256 outputValue = _minOutputAmount * outputPrice.value;
+
+        Require.that(
+            outputValue > inputValue - (inputValue * slippageMinimum() / _SLIPPAGE_BASE),
```

Comment:
I did a double take here. I think we want to do `outputValue < ...` because we want the output value to be sufficiently below the slippage minimum.

Recall the reason for us having this is we want to make sure the output amount has a sufficient buffer in case the price drops between the settlement of the GM tokens